src/solutions/solution_interface.jl

```diff
@@ -54,10 +54,51 @@ SymbolicIndexingInterface.is_time_dependent(::AbstractTimeseriesSolution) = true
 
 SymbolicIndexingInterface.is_time_dependent(::AbstractNoTimeSolution) = false
 
+function SymbolicIndexingInterface.parameter_timeseries(A::AbstractTimeseriesSolution)
+    if !isdefined(A, :discretes) || (discretes = A.discretes) === nothing
+        return [0]
+    end
+    return discretes.t
+end
+
+function SymbolicIndexingInterface.parameter_values_at_time(A::AbstractTimeseriesSolution, t)
+    ps = parameter_values(A)
+    if !isdefined(A, :discretes) || (discretes = A.discretes) === nothing
+        return ps
+    end
+    return SciMLStructures.replace(ps, SciMLStructures.Discrete(), discretes.u[t])
+end
+
+function SymbolicIndexingInterface.parameter_values_at_state_time(A::AbstractTimeseriesSolution, tidx)
+    ps = parameter_values(A)
+    if !isdefined(A, :discretes) || (discretes = A.discretes) === nothing
+        return ps
+    end
+    t = A.t[tidx]
+    idx = searchsortedfirst(discretes.t, t)
```

Comment:
`searchsortedfirst` returns the first index greater than or equivalent to `t`, that means that the case where there is an exact match `t` has to be handled separately, since then you should return `discretes.u[idx]` instead of `discretes.u[idx - 1]`?

Reply:
Thanks for pointing that out! I'll fix it

Reply:
Fixed in https://github.com/SciML/SciMLBase.jl/pull/645/commits/1533e0f53d12043128b46954c68319828d709e01